nitro-src/claim.ts

```diff
@@ -38,6 +44,40 @@ export function claim(
     )
       throw Error;
 
+    let surplus = BigNumber.from(initialHoldings[assetIndex]);
+    // Any guarantees before this one have priority on the funds
+    // So we must account for that by reducing the surplus
+    for (
+      let guaranteeIndex = 0;
+      guaranteeIndex < targetChannelIndex;
+      guaranteeIndex++
+    ) {
+      const { amount } = guarantees[guaranteeIndex];
+
+      surplus = surplus.sub(min(BigNumber.from(amount), surplus)); // Prevent going below 0
+    }
+    // If there are not enough funds to fund the guarantee we return immediately
+    if (surplus.lte(0)) {
+      return {
+        updatedGuaranteeOutcome,
+        updatedHoldings,
+        updatedTargetOutcome,
+        exit,
+      };
+    } else {
+      const currentAmount = BigNumber.from(
+        updatedGuaranteeOutcome[assetIndex].allocations[targetChannelIndex]
+          .amount
+      );
+      const newAmount = currentAmount
+        .sub(min(surplus, currentAmount)) // It's possible that surplus is large
+        .toHexString();
+
+      // If we do have enough funds we update the guarantee to indicate they have been allocated
+      updatedGuaranteeOutcome[assetIndex].allocations[
+        targetChannelIndex
+      ].amount = newAmount;
+    }
```

Comment:
I'm not sure this is quite right. 

Imagine if the guarantor channel `L` (which is really a ledger channel in this new approach) has a holdings of `10` coins, and has a single allocation guaranteeing `J` with `10` coins and priority `AIX`. If the outcome of `J` is `[A:5,B:5,I:10]`, and we claim on `L` specifying that we want to pay out `A`, then `A` should get `5` coins. 

In this scenario the holdings of `L` only decreases by `5`. So the guarantee amount for `J` should be decreased by `5`, not by `10`.

Suggestion: move the decrement of  `updatedGuaranteeOutcome[assetIndex].allocations[
        targetChannelIndex
      ]` to  L118, and use the value `affordsForDestination` rather than `currentAmount.sub(surplus)`.